src/lib/token.ts

```diff
@@ -0,0 +1,69 @@
+import { Config, TokenSearchRules, TokenOptions } from '../types'
+import crypto from 'crypto'
+
+function encode64(str: any) {
+  return Buffer.from(JSON.stringify(str)).toString('base64')
+}
+
+function sign(apiKey: string, encodedHeader: string, encodedPayload: string) {
+  return crypto
+    .createHmac('sha256', apiKey)
+    .update(`${encodedHeader}.${encodedPayload}`)
+    .digest('base64')
+    .replace(/\+/g, '-')
+    .replace(/\//g, '_')
+    .replace(/=/g, '')
+}
+
+function createHeader() {
+  const header = {
+    alg: 'HS256',
+    typ: 'JWT',
+  }
+
+  return encode64(header).replace(/=/g, '')
+}
+
+function createPayload(
+  searchRules: TokenSearchRules,
+  apiKey: string,
+  expiresAt: Date | null
+) {
+  const payload = {
+    exp: expiresAt?.getTime() || null,
```

Comment:
I'm not sure if you need to keep the `|| null`, because `null` and `undefined` will have the same result:

```js
console.log(JSON.stringify({ ab: undefined }))
> "{}"
console.log(JSON.stringify({ ab: null }))
> '{"ab":null}'
```

And if `exp` does not have a valid value we will just assume it does not expires.

Reply:
Yes I kept it so that the user was aware that they didn't put a limit! But it may be overkill :) I removed it